### ISSUE TITLE
Usability tests

### ISSUE THREAD

**Issue by ManfredKarrer** [bisq-network/bisq#94]

**Comment by ManfredKarrer** [comment]
Here is a summary of the feedback of the usability tests:
The general result was that it seems that the app is now usable even for not experienced Bitcoiners but there is of course  a lot of headroom for improvements.

The main points was:
- We should use more explicit text. Would be more friendly and less technical.
- People does not read the small info text, nearly nobody clicked on read more or open help.
- Min. amount is not clear yet.
- One very important feedback was to remove the global balance field and make the dedicated wallet concept clear: Pay in BTC on demand and in context and pay out as soon it is available. The withdrawal screen will be just used exceptional (if user want to postpone his payout for some reason)

I implemented the most important issues, but don't want to spend too much time yet for implementing the remaining ones.
There are some known areas like the Popups which are not covered here as well as the not updated screens like (funds, arbiters).
I will add issues for clear defined tasks but some points in the feedback/recommendations will need more discussion.

I will only add here the points which are not already fixed/implemented.
@Lloyd and Joachim: Feel free to add/correct anything missing (if it is not already implemented in the current version)

Test 1 with Maria (never used Bitcoin, she is a non technical user):
That was with the older version before starting the latest UI iteration.

Create offer screen:
- Meaning of Min. amount not clear
- Question mark for deposit amount details not recognized as roll over
- Deposit amount not clear why she need to pay in something
- Recommended to add fiat currency to btc values specially at fees (0.0002 BTC / 0.14 EUR)
- Recommended to pre-fill price with average market price, did not have any idea which price she should use
- Address not recognized that it is a hyper link opening the wallet

Test 2 with Joachim (experienced Bitcoiner):
Joachim has helped already a few times with usability feedback from the very first version on.

Account setup:
- If an entry is added to the list in the restrictions screen, it should scroll and select the new entry in the list to give the user a visual feedback
- Payment account screen: remove "manage accounts" and only show an "add account" button
- Recommended to remove sub-title (explanations) in the left wizard buttons
- Explain why registration fee is needed: "For security reasons, all Bitquare accounts need to be registered in the Blockchain which requires a payment of 0.0002 BTC."
- Step 1: "Please transfer exactly 0.0002 to your Bitsquare wallet. Button: "open external wallet" (Implementation: open QT with data filled or if not installed pop-up with all the information)
  If exactly 0.002 BTC received --> Popup: You have sufficient BTC to register your Bitsquare account in the Blockchain. Button "Register and pay 0.0002 BTC"
  If more than 0.002 BTC received --> Popup: You have sufficient BTC to register your Bitsquare account in the Blockchain. 
  Show Change address: XXXXXXXXXXXXXXXX
  Button "Register and pay 0.0002 BTC and refund 0.9998 BTC to the change address"
- Popup after registration payment: Replace "copy" button with hyper link (gives it less weight) 
- Instead of the bitcoin address display hyperlink "open external wallet"

Orderbook:
- remove filter and make it optional with button top left over the table
- move "create new offer" button top right over the table

Create offer: 
- Replace question mark icon with hyper link and clear text
- Use "mining fee" instead of BTC network fee
- Recommendation: instead of "Amount in BTC" x "Price" = "Amount in Fiat" input fields he recommends a more textual version:
- Recommendation: Use kind of step wizard.
  Step 1: How many BTC do you want to buy?
  Step 2:  "Please transfer exactly 0.8003 BTC to a dedicated trade-wallet. Button: "open external wallet" (Implementation: open QT with data filled or if not installed pop-up with all the information)
  Step 3:
  If exactly 0.8003 BTC received --> Popup: You have sufficient BTC to place your offer. Button "Place offer and pay trading fee"
  If more than 0.8003 BTC received --> Popup: You have sufficient BTC to place your offer. Button "Place offer and pay trading fee"
  Show Change address: XXXXXXXXXXXXXXXX
  Button "Place offer and pay trading fee and refund 0.9998 BTC to the change address"

Take offer:
- if min.amount== amount don't display it min.amount field
- Recommendation: User "accept offer" instead of "take offer"

Trade process:
- When bank tx done display checkbox: "I confirm that I have made the payment of xxx EUR"
- When Fiat received add checkbox explicitly confirming: "I confirm that I received xxx EUR on my Bank account", after confirming display "Payout" button
- Use more explicit and friendly text for every step
- replace tx id with hyper link to block explorer

Test 3 with Roman (no experience with Bitcoin but he has read our whitepaper):
- At startup was not sure what he needs to do, searched for most logical solution -> account. We should display a info to first time user that he can play around without registration, just when he wants to trade he need to register.
- Problems at registration, restrictions screen not clear as well registration fee payment
- Min. amount not clear
- Did not found funds/withdrawal page 
- tried to type the first letter to jump to the entry in the dropdown list, but not supported in java out of the box

Test 4 with Lloyd (experienced Bitcoiner):
Account setup:
- Password validation missing
- Selecting 2 times the same entry to the list (restrictions) does not give visual feedback
- If a entry is added to the list in the restrictions screen, it should scroll and select the new entry in the list to give the user a visual feedback
- tried to type the first letter to jump to the entry in the dropdown list, but not supported in java out of the box
- some countries missing (North Korea, Guernsey, Jersey, Isle of Man...)
- Bug: Continue button in restrictions screen not disabled when removing arbiter
- Bug: adding a 2. bank account and then click "continue to next step" does not save the bank account
- multiple bank accounts with same bank have same name atm. should be either combination with currency/account nr or open dialog with input for user defined name
- Move accepted bank accounts from restrictions to payment accounts screen
- Rename restrictions to arbiters selection (accepted languages will be removed anyway probably)
- Question: Is payment label used to open/pre fill external wallet added to tx and public visible in blockchain? I assume it is not, but need to check that, if so we should not use the label. Using a label should be optional (preferences)

Orderbook:
- Recommendation: Display all currencies in orderbook. use filter to only display currency of selected bank account

Create offer:
- tab must be not closable after money has been added
- Bug: Changing bank account not updated in details section
- Bank account selection should be more explicit directly in create/take offer screen, combobox menu in main menu is not explicit enough
- Recommendation: If user pay in too much that will be added automatically as additional deposit increasing the "reputation" of the trader. Good suggestion which we should use as additional protection possibility.

Trade process:
- Nr. of confirmation should be displayed explicit with text, spinner/confirmation component is not enough
- Explicit text on top in each step, small info text is not enough, info text in stepper is also not enough
- Bug: copy amount include EUR text, should be only amount value
- Display mining fee explicit at withdrawal

**Comment by notmandatory** [comment]
I support just removing the "Min Amount" part of the order process for now.  It will simplify the UI and I don't think it's needed for an initial release.

**Comment by btcll** [comment]
I think this captures my feedback nicely! Thanks for writing it all up Manfred.
